anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -198,46 +261,102 @@ def update_counts(self):
             logger.error("Could not update group counts")
             raise
 
-    def _load_mapper(self, group_obj):
+    def _load_mapper(self, group_obj) -> FeedDataMapper:
         """
         Find and instantiate the right mapper object for the given group.
 
         :param group_obj:
-        :return:
+        :return: feed data mapper for this group
+        :rtype: FeedDataMapper
         """
-        if not hasattr(self.__class__.__group_data_mappers__, "get"):
-            mapper = self.__class__.__group_data_mappers__
-        else:
-            mapper = self.__class__.__group_data_mappers__.get(group_obj.name)
+        mapper = self.__class__.__group_data_mappers__.get(group_obj.name)
 
         if not mapper:
             raise Exception(
                 "No mapper class found for group: {}".format(group_obj.name)
             )
 
-            # If it's a class, instantiate it
-        if type(mapper) == type:
-            mapper = mapper(self.__feed_name__, group_obj.name, keyname=None)
-
         return mapper
 
+    def _process_group_file_records(
+        self,
+        db: Session,
+        group_download_result: GroupDownloadResult,
+        group_obj: FeedGroupMetadata,
+        local_repo: Optional[LocalFeedDataRepo],
+    ) -> int:
+        """
+        Convert the download results for a feed group into database records and write to the database session
+        Does not flush, so transaction can be rolled back if an exception is encountered.
+        :param db: sqlalchemy database session
+        :type db: Session
+        :param group_download_result: group download result record to update
+        :type group_download_result: GroupDownloadResult
+        :param group_obj: metadata record for this feed group
+        :type group_obj: FeedGroupMetadata
+        :param local_repo: LocalFeedDataRepo (disk cache for download results)
+        :type local_repo: Optional[LocalFeedDataRepo], defaults to None
+        :return: int
+        :rtype: total number of records updated
+        """
+        mapper = self._load_mapper(group_obj)
```

Comment:
Would be good to have some comments describing the logic flow here.

Reply:
done